pyomo/contrib/parmest/parmest.py

```diff
@@ -433,14 +434,69 @@ def _Q_opt(self, ThetaVals=None, solver="ef_ipopt",
                                     EF_name = "_Q_opt",
                                     suppress_warnings=True,
                                     scenario_creator_kwargs=scenario_creator_options)
+
+        # if pre_solve = True, solve square problem with parameters to be fitted
+        # fixed before solving parameter estimation problem
+        if pre_solve:
+            # Fix fitted parameters for initialization
+            for i, theta in enumerate(self.theta_names):
+                # Use parser in ComponentUID to locate the component
+                var_cuid = ComponentUID(theta)
+                var_validate = var_cuid.find_component_on(self.parmest_model)
+                if var_validate is None:
+                    logger.warning(
+                        "theta_name[%s] (%s) was not found on the model",
+                        (i, theta))
+                else:
+                    try:
+                        # If the component that was found is not a variable,
+                        # this will generate an exception (and the warning
+                        # in the 'except')
+                        var_validate.fix()
+                        # We want to standardize on the CUID string
+                        # representation
+                        # self.theta_names[i] = repr(var_cuid)
+                    except:
+                        logger.warning(theta + ' is not a variable')
+
+            init_solver = SolverFactory('ipopt')
+            if self.solver_options is not None:
+                for key in self.solver_options:
+                    init_solver.options[key] = self.solver_options[key]
+
+            # Initial solve of square problem with ipopt
+            # Fitted parameters are fixed
+            init_solver.solve(ef, tee = True)
+
+            # Un-fix fitted parameters for estimation
+            for i, theta in enumerate(self.theta_names):
+                # Use parser in ComponentUID to locate the component
+                var_cuid = ComponentUID(theta)
+                var_validate = var_cuid.find_component_on(self.parmest_model)
+                if var_validate is None:
+                    logger.warning(
+                        "theta_name[%s] (%s) was not found on the model",
+                        (i, theta))
+                else:
+                    try:
+                        # If the component that was found is not a variable,
+                        # this will generate an exception (and the warning
+                        # in the 'except')
+                        var_validate.unfix()
+                        # We want to standardize on the CUID string
+                        # representation
+                        # self.theta_names[i] = repr(var_cuid)
+                    except:
+                        logger.warning(theta + ' is not a variable')
```

Comment:
Instead of repeating this logic I think it would make more sense to just keep a list of the vars being fixed above and then just iterate over that list to unfix them. Going through the process of searching for them again is inefficient and there is no need to redo the validation logic since the error would have already been thrown above.